INSTALL.md

```diff
@@ -24,56 +20,142 @@
+   ```
+3. Check docker-compose.yml contents are suitable for your needs. It will be ok as is for most people
+4. Update the environment variables in stack.env according to your needs. As a minimum enter the VIN of your car, and the connection details for your MQTT server. If you want BLE detection enter the BLE MAC address of the car (see below for instructions on how to find this TODO):
+   ```yaml
+   # Optional for car presence detection; If multiple cars, separate with , or | or white space
+   #
+   BLE_MAC_LIST=
+   # Default 5 (seconds)
+   #
+   BLE_CMD_RETRY_DELAY=
+   ### Default false
+   #
+   DEBUG=
+   # Hostname or IP address
+   #
+   MQTT_SERVER=
+   # Service port # or name
+   #
+   MQTT_PORT=1883
+   # If no username provided, anonymous mode.
+   #
+   MQTT_USERNAME=
+   # If you have special characters, wrap with ' at both ends; escape ' if needed
+   #
+   MQTT_PASSWORD=
+   # Default 120 (seconds)
+   #
+   PRESENCE_DETECTION_LOOP_DELAY=
+   # Default 240 (seconds)
+   #
+   PRESENCE_DETECTION_TTL=
+   # Mandatory; if multiple VINs separate with , or | or white space
+   #
+   VIN_LIST=
+   # Your timezone
+   # Ref: https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+   #
+   TZ='Europe/London'
+   #
+   # WARNING; If you run Home Assistant, keep this true unless you know what you're doing
+   #
+   ENABLE_HA_FEATURES=true
+   ```
+5. Create the Docker volume: `docker volume create tesla_ble_mqtt`
+6. Create a symbolic link to the environment file: `ln -s stack.env env`
+7. Start the container: `docker compose up -d`
+8. Check the logs `docker logs tesla_ble_mqtt`. Typical logs after start up look like this (when DEBUG=false):
 ```
-tesla_ble_mqtt_docker by Iain Bullock 2024 https://github.com/iainbullock/tesla_ble_mqtt_docker
-Inspiration by Raphael Murray https://github.com/raphmur
-Instructions by Shankar Kumarasamy https://shankarkumarasamy.blog/2024/01/28/tesla-developer-api-guide-ble-key-pair-auth-and-vehicle-commands-part-3
 Configuration Options are:
-TESLA_VIN=****
-MQTT_IP=192.168.1.51
-MQTT_PORT=1883
-MQTT_USER=****
-MQTT_PWD=Not Shown
-Setting up auto discovery for Home Assistant
-Listening to MQTT
+  BLE_CMD_RETRY_DELAY=5
+  BLE_MAC_LIST=40:XX:XX:XX:XX:F9
+  DEBUG=false
+  MQTT_SERVER=192.168.1.5
+  MQTT_PORT=1883
+  MQTT_PASSWORD=Not Shown
+  MQTT_USERNAME=XXXXXX
+  PRESENCE_DETECTION_LOOP_DELAY=120
+  PRESENCE_DETECTION_TTL=240
+  VIN_LIST=LRWXXXXXXXXXXX403
+  ENABLE_HA_FEATURES=true
+Setting up MQTT clients with authentication
+Presence detection is enable with a TTL of 240 seconds
+Setting up HA auto discovery for vin LRWXXXXXXXXXXX403
+Discarding any unread MQTT messages for LRWXXXXXXXXXXX403
+Listening for Home Assistant Start (in background)
+Entering main loop...
+Lauching background listen_to_mqtt_loop...
+Entering Listen to MQTT loop...
+Launch BLE scanning for car presence every 120 seconds
+Launching listen_to_mqtt

```

Comment:
I wonder, this logging will change as we improve _core...
Maybe we need to log a message like "All set, ready to go" ... that will always be there after initialization

Reply:
I will change it in future if we implement a message like "All set, ready to go" . In the meantime I will leave it as is